nupic/research/frameworks/backprop_structure/networks/vdrop_lenet.py

```diff
@@ -0,0 +1,203 @@
+                    input_shape[0],
+                    cnn_out_channels[0],
+                    kernel_size,
+                    vdrop_data,
+                    target_density=conv_target_density[0],
+                ),
+            ),
+            ("cnn1_maxpool", nn.MaxPool2d(maxpool_stride)),
+        ]
+
+        if use_batch_norm:
+            modules.append(
+                (
+                    "cnn1_bn",
+                    nn.BatchNorm2d(
+                        cnn_out_channels[0],
+                        affine=False,
+                        track_running_stats=bn_track_running_stats,
+                    ),
+                )
+            )
+
+        modules += [
+            ("cnn1_relu", nn.ReLU(inplace=True)),
+            # -------------
+            # Conv Block
+            # -------------
+            (
+                "vdrop_cnn2",
+                prunable_vdrop_conv2d(
+                    cnn_out_channels[0],
+                    cnn_out_channels[1],
+                    kernel_size,
+                    vdrop_data,
+                    target_density=conv_target_density[1],
+                ),
+            ),
+            ("cnn2_maxpool", nn.MaxPool2d(maxpool_stride)),
+        ]
+
+        if use_batch_norm:
+            modules.append(
+                (
+                    "cnn2_bn",
+                    nn.BatchNorm2d(
+                        cnn_out_channels[1],
+                        affine=False,
+                        track_running_stats=bn_track_running_stats,
+                    ),
+                )
+            )
+
+        modules += [
+            ("cnn2_relu", nn.ReLU(inplace=True)),
+            ("flatten", nn.Flatten()),
+            # -------------
+            # Linear Block
+            # -------------
+            (
+                "vdrop_fc1",
+                prunable_vdrop_linear(
+                    (feature_map_sidelength ** 2) * cnn_out_channels[1],
+                    linear_units,
+                    vdrop_data,
+                    target_density=linear_target_density[0],
+                ),
+            ),
+        ]
+        if use_batch_norm:
+            modules.append(
+                (
+                    "fc1_bn",
+                    nn.BatchNorm1d(
+                        linear_units,
+                        affine=False,
+                        track_running_stats=bn_track_running_stats,
+                    ),
+                )
+            )
+
+        modules += [
+            ("fc1_relu", nn.ReLU(inplace=True)),
+            # -------------
+            # Output Layer
+            # -------------
+            (
+                "vdrop_fc2",
+                prunable_vdrop_linear(
+                    linear_units,
+                    num_classes,
+                    vdrop_data,
+                    target_density=linear_target_density[1],
+                ),
+            ),
+        ]
+        self.classifier = nn.Sequential(OrderedDict(modules))
```

Comment:
To improve checkpoint compatibility with our other GSC networks, I think it would be better for this class to inherit from `nn.Sequential`. Otherwise `*.classifier.*` is going to appear in every state_dict key, and it will be harder for us to export parameters to a static SparseWeights model. After changing to inherit from `nn.Sequential`, this line would change to `super().__init__(OrderedDict(modules))`.